src/renderer/index.jsx

```diff
@@ -127,34 +79,87 @@ const start = async () => {
       }
       teams[t.index].order = order;
     });
-    teamConfigChange(teams);
+    await this.teamConfigChange(teams);
     return teamIndex;
+  };
+
+  teamConfigChange = async (updatedTeams, callback) => {
+    const updatedConfig = await ipcRenderer.invoke(UPDATE_TEAMS, updatedTeams);
+    await this.reloadConfig();
+    if (callback) {
+      callback(updatedConfig);
+    }
+  };
+
+  showBadge = (sessionExpired, unreadCount, mentionCount) => {
+    ipcRenderer.send('update-unread', {
+      sessionExpired,
+      unreadCount,
+      mentionCount,
+    });
+  }
+
+  reloadConfig = async () => {
+    const config = await this.requestConfig();
+    this.setState({config});
+  };
+
+  requestConfig = async (exitOnError) => {
+    // todo: should we block?
+    try {
+      console.log('requested configuration');
+      const configRequest = await ipcRenderer.invoke(GET_CONFIGURATION);
+      console.log(`config is: ${JSON.stringify(configRequest)}`);
+      return configRequest;
+    } catch (err) {
+      console.log(`there was an error with the config: ${err}`);
+      if (exitOnError) {
+        ipcRenderer.send(QUIT, `unable to load configuration: ${err}`, err.stack);
+      }
+    }
+    return null;
+  };
+
+  getInitialIndex = (teamList) => {
+    if (teamList) {
+      const element = teamList.find((e) => e.order === 0);
+      return element ? teamList.indexOf(element) : 0;
+    }
+    return 0;
+  }
+
+  openMenu = () => {
+    if (process.platform !== 'darwin') {
+      ipcRenderer.send('open-app-menu');
+    }
   }
 
-  const component = (
-    <MainPage
-      teams={teams}
-      localTeams={config.localTeams}
-      initialIndex={initialIndex}
-      onBadgeChange={showBadge}
-      onTeamConfigChange={teamConfigChange}
-      useSpellChecker={config.useSpellChecker}
-      deeplinkingUrl={deeplinkingUrl}
-      showAddServerButton={config.enableServerManagement}
-      moveTabs={moveTabs}
-      openMenu={openMenu}
-      darkMode={config.darkMode}
-      appName={config.appName}
-    />);
-
-  ReactDOM.render(
-    component,
-    document.getElementById('app')
-  );
-};
-
-// Deny drag&drop navigation in mainWindow.
-// Drag&drop is allowed in webview of index.html.
-document.addEventListener('dragover', (event) => event.preventDefault());
-document.addEventListener('drop', (event) => event.preventDefault());
-start();
+  render() {
+    const {config, initialIndex, deeplinkingUrl} = this.state;
+    if (!config) {
+      return null;
```

Comment:
out of the scope of this PR, but we should render something in the screen in case something goes wrong. Some sort of spinning loader with a "Loading configuration..." sort of screen

Reply:
That's a good idea, we can probably rope in Dean's new loading spinner for this.

Reply:
yeah, I thought about it, once we get it to work :)